File: functions/src/models/reader.ts
export class Reader {
    id?: string;
    email?: string;
    name?: string;
    phoneNumber?: string;
    phoneNumberAlt?: string;
  
    constructor(
      id?: string,
      email?: string,
      name?: string,
      phoneNumber?: string,
      phoneNumberAlt?: string
    ) {
      this.id = id;
      this.email = email;
      this.name = name;
      this.phoneNumber = phoneNumber;
      this.phoneNumberAlt = phoneNumberAlt;
    }
  
    static fromJson(json: any): Reader {
      return new Reader(
        json['id'],
        json['email'],
        json['name'],
        json['phoneNumber'],
        json['phoneNumberAlt']
      );
    }
  
    toJson(): object {
      return {
        id: this.id ?? null,
        email: this.email ?? null,
        name: this.name ?? null,
        phoneNumber: this.phoneNumber ?? null,
        phoneNumberAlt: this.phoneNumberAlt ?? null
      };
    }
  
    static fromFirestore(data: FirebaseFirestore.DocumentData): Reader {
      return new Reader(
        data['id'],
        data['email'],
        data['name'],
        data['phoneNumber'],
        data['phoneNumberAlt']
      );
    }
  }
  

File: lib/data/models/reader.dart
class Reader {
  final String id;
  final String email;
  final String name;
  final String phoneNumber;
  final String phoneNumberAlt;

  Reader({
    required this.id,
    required this.email,
    required this.name,
    required this.phoneNumber,
    required this.phoneNumberAlt,
  });

  factory Reader.fromMap(Map<String, dynamic> json, String docId) {
    print("ReaderMapdoc = $docId $json ");
    return Reader(
      id: docId,
      email: json['email'] ?? '',
      name: json['name'] ?? '',
      phoneNumber: json['phoneNumber'] ?? '',
      phoneNumberAlt: json['phoneNumberAlt'] ?? '',
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'email': email,
      'name': name,
      'phoneNumber': phoneNumber,
      'phoneNumberAlt': phoneNumberAlt,
    };
  }
}
